db/sqlc/store.go

```diff
@@ -93,37 +93,37 @@ func (store *Store) TransferTx(ctx context.Context, arg TransferTxParams) (Trans
 		}
 
 		// Update accounts' balance
-		account1, err := q.GetAccountForUpdate(ctx, arg.FromAccountID)
-		if err != nil {
-			return err
-		}
+		if arg.FromAccountID < arg.ToAccountID {
+			result.FromAccount, result.ToAccount, err = depositAccount(ctx, q, arg.FromAccountID, -arg.Amount, arg.ToAccountID, arg.Amount)
+		} else {
+			result.ToAccount, result.FromAccount, err = depositAccount(ctx, q, arg.ToAccountID, arg.Amount, arg.FromAccountID, -arg.Amount)
```

Comment:
Is there any reason to add this?

Reply:
We are checking the query order to make sure they are going with a consistent order. This is the key to avoiding deadlock while processing transactions.  Hence we are making sure the smaller account id goes first.